airbyte-integrations/connectors/source-hubspot/source_hubspot/api.py

```diff
@@ -460,6 +460,45 @@ class ContactListStream(Stream):
     limit_field = "count"
 
 
+class DealStageHistoryStream(Stream):
+    """Deal stage history, API v1
+    Deal stage history is exposed by the v1 API, but not the v3 API.
+    The v1 endpoint requires the contacts scope.
+    Docs: https://legacydocs.hubspot.com/docs/methods/deals/get-all-deals
+    """
+    
+    url = "/deals/v1/deal/paged"
+    more_key = "hasMore"
+    data_field = "deals"
+    updated_at_field = "timestamp"
+    
+    def _transform(self, records: Iterable) -> Iterable:
+        for record in super()._transform(records):
+            yield {"id": record.get("dealId"), **record.get("properties", {}).get("dealstage")}
```

Comment:
I ran the script and it crashes here. Not all records of this stream have `dealstage`, which means that we do not have the `updated_at_field` variable and the `read` method breaks.
```Traceback (most recent call last):
  File "/airbyte/airbyte-integrations/connectors/source-hubspot/main_dev.py", line 33, in <module>
    launch(source, sys.argv[1:])
  File "/airbyte/airbyte-integrations/bases/base-python/base_python/entrypoint.py", line 121, in launch
    AirbyteEntrypoint(source).start(args)
  File "/airbyte/airbyte-integrations/bases/base-python/base_python/entrypoint.py", line 112, in start
    for message in generator:
  File "/airbyte/airbyte-integrations/bases/base-python/base_python/source.py", line 89, in read
    yield from self._read_stream(logger=logger, client=client, configured_stream=configured_stream, state=total_state)
  File "/airbyte/airbyte-integrations/bases/base-python/base_python/source.py", line 108, in _read_stream
    for record in client.read_stream(configured_stream.stream):
  File "/airbyte/airbyte-integrations/bases/base-python/base_python/client.py", line 91, in read_stream
    for message in method(fields=fields):
  File "/airbyte/airbyte-integrations/connectors/source-hubspot/source_hubspot/api.py", line 496, in list
    for record in self._stage_history.list(fields):
  File "/airbyte/airbyte-integrations/connectors/source-hubspot/source_hubspot/api.py", line 484, in list
    yield from self.read(partial(self._api.get, url=self.url), params)
  File "/airbyte/airbyte-integrations/connectors/source-hubspot/source_hubspot/api.py", line 293, in read
    yield from self._filter_dynamic_fields(self._filter_old_records(self._transform(self._read(getter, params))))
  File "/airbyte/airbyte-integrations/connectors/source-hubspot/source_hubspot/api.py", line 225, in _filter_dynamic_fields
    for record in records:
  File "/airbyte/airbyte-integrations/connectors/source-hubspot/source_hubspot/api.py", line 252, in _filter_old_records
    updated_at = record[self.updated_at_field]
KeyError: 'timestamp'

Process finished with exit code 1
```

Reply:
Thank you for the trace, [updated to filter for records that have dealstage](https://github.com/airbytehq/airbyte/pull/3734/files#diff-37c875e03f2e65ea1423534b0b634e18eaca797135f8d9df32360407b1d0ccb9R479) (I tried locally testing with `python -m pytest -p integration_tests.acceptance` and dev hubspot creds but the integration tests timed out so I need to tweak my environment)